fedbiomed/researcher/secagg/_secagg_context.py

```diff
@@ -538,3 +599,112 @@ def _payload_create(self) -> Tuple[Union[dict, None], bool]:
         # end dummy payload
 
         return context, True
+
+
+class SecaggDhContext(SecaggContext):
+    """
+    Handles a Secure Aggregation Diffie Hellman context element on the researcher side.
+    """
+
+    def __init__(self, parties: List[str], experiment_id: str, secagg_id: Union[str, None] = None):
+        """Constructor of the class.
+
+        Args:
+            parties: list of parties participating in the secagg context element setup, named
+                by their unique id (`node_id`, `researcher_id`).
+                There must be at least 3 parties, and the first party is this researcher
+            experiment_id: ID of the experiment to which this secagg context element is attached.
+            secagg_id: optional secagg context element ID to use for this element.
+                Default is None, which means a unique element ID will be generated.
+
+        Raises:
+            FedbiomedSecaggError: bad argument type or value
+        """
+        super().__init__(parties, experiment_id, secagg_id)
+
+        if len(parties) < 2:
+            raise FedbiomedSecaggError(
+                f'{ErrorNumbers.FB415.value}: LOM, bad parameter `parties` : {parties} : need'
+                'at least 3 parties for secure aggregation')
```

Comment:
Two points here:
1. error message is not consistent with the check: if `len(parties)  = 2` error won't raise
2. this part of code is not consistent with what has been done in `node` 's `secagg` (`fedbiomed.node.secagg._secagg_context`) where there is a `_min_num_parties` attribute available as a constant and used within class implementation. To fix that, I think we should create a specific enum class in `common.constants`

Reply:
>  error message is not consistent with the check: if `len(parties)  = 2` error won't raise
Ok sorry, my mistake, parties are filtered (researcher is removed) beforehand. But second point still holds

Reply:
Hi @ybouilla,

I don't know what happed but `min_num_parties` is not existing anymore. I remember that I created this constant variable. Actually it is not a constant it is variable attached to context type DH or Shamir.

Reply:
Hi @srcansiz 
Unless I'm mistaking, you introduced the `_min_num_parties` for node but not for researcher, this is what Yannick is spotting. Does not look like some code was reverted.

Reply:
Thank you @mvesin, I miss understood the comment.